lib/main.rb

```diff
@@ -0,0 +1,82 @@
+require_relative 'planet'
+require_relative 'solar_system'
+
+# Provide details about the planet, called in user selection #2
+def planet_details(solar_system)
+  print "Enter the planet name you would like to learn about: "
+  # you, user_namechoice is a string
+  user_namechoice = gets.chomp.capitalize
+  solar_system.planets.each do | planet |
+    if planet.name == user_namechoice
+      found_planet = solar_system.find_planet_by_name(planet)
+      puts found_planet.summary
```

Comment:
This loop should live inside the `SolarSystem#find_planet_by_name`  method. Then, this code would look like:

```ruby
def planet_details(solar_system)
  print "Planet name: "
  planet_name = gets.chomp
  planet = solar_system.find_planet_by_name(planet_name)
  puts planet.summary
end
```

You also might want to include code for what to do if the planet wasn't found.